spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/source/TestMetadataTableReadableMetrics.java

```diff
@@ -0,0 +1,498 @@
+                "1",
+                DateTimeUtil.dateFromDays(1),
+                DateTimeUtil.timeFromMicros(1),
+                DateTimeUtil.timestampFromMicros(1L),
+                Base64.getDecoder().decode("1111"),
+                ByteBuffer.wrap(Base64.getDecoder().decode("1111"))),
+            createPrimitiveRecord(
+                true,
+                2,
+                2L,
+                0,
+                2.0D,
+                new BigDecimal("2.00"),
+                "2",
+                DateTimeUtil.dateFromDays(2),
+                DateTimeUtil.timeFromMicros(2),
+                DateTimeUtil.timestampFromMicros(2L),
+                Base64.getDecoder().decode("2222"),
+                ByteBuffer.wrap(Base64.getDecoder().decode("2222"))));
+
+    DataFile dataFile =
+        FileHelpers.writeDataFile(table, Files.localOutput(temp.newFile()), records);
+    table.newAppend().appendFile(dataFile).commit();
+
+    Dataset<Row> nestedSelect =
+        spark
+            .read()
+            .format("iceberg")
+            .load("default." + tableName + ".files")
+            .select(
+                "readable_metrics.intCol.lower_bound", "readable_metrics.booleanCol.upper_bound");
+
+    List<Object[]> metricSelectResult = rowsToJava(nestedSelect.collectAsList());
+    assertEquals("Rows do not match", ImmutableList.of(row("1", "true")), metricSelectResult);
+
+    Dataset<Row> mixedSelect =
+        spark
+            .read()
+            .format("iceberg")
+            .load("default." + tableName + ".files")
+            .select("file_path", "readable_metrics.longCol.value_count");
+
+    List<Object[]> mixedSelectResult = rowsToJava(mixedSelect.collectAsList());
+    assertEquals(
+        "Rows do not match",
+        ImmutableList.of(row(dataFile.path().toString(), 2L)),
+        mixedSelectResult);
+
+    Dataset<Row> mixedSelect2 =
+        spark
+            .read()
+            .format("iceberg")
+            .load("default." + tableName + ".files")
+            .select("sort_order_id", "readable_metrics.longCol.value_count");
+
+    List<Object[]> mixedSelectResult2 = rowsToJava(mixedSelect2.collectAsList());
+    assertEquals("Rows do not match", ImmutableList.of(row(0, 2L)), mixedSelectResult2);
+
+    Dataset<Row> reversedSelect =
+        spark
+            .read()
+            .format("iceberg")
+            .load("default." + tableName + ".files")
+            .select("readable_metrics.longCol.value_count", "file_path");
+    List<Object[]> reversedSelectResult = rowsToJava(reversedSelect.collectAsList());
+    assertEquals(
+        "Rows do not match",
+        ImmutableList.of(row(2L, dataFile.path().toString())),
+        reversedSelectResult);
+  }
+
+  @Test
+  public void testNullNanValues() throws Exception {
+    String tableName = "testNullNanValues";
+    Table table = createTable(tableName, PRIMITIVE_SCHEMA, PartitionSpec.unpartitioned());
+
+    List<Record> records =
```

Comment:
This record could just be added to the base table all the tests use as well right?

Reply:
Yea, done, removed the extra nullNan test and do all the checks in the primitive test